gorm/transaction.go

```diff
@@ -45,10 +56,41 @@ func FromContext(ctx context.Context) (txn *Transaction, ok bool) {
 type Transaction struct {
 	mu              sync.Mutex
 	parent          *gorm.DB
+	parentRO        *gorm.DB
 	current         *gorm.DB
+	txOpts          *sql.TxOptions
```

Comment:
Consider using e.g. `currentOpts *databaseOptions` rather than separately `txOpts` and `currentDB` etc.

I think you could more directly compare "current and prior options" that way; just keep the options value from when the transaction was started and compare options to options.